libbeat/testing/seed.go

```diff
@@ -0,0 +1,39 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package testing
+
+import (
+	"flag"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+var (
+	SeedFlag = flag.Int64("seed", 0, "Randomization seed")
+)
+
+func SeedPRNG(t *testing.T) {
```

Comment:
Do not use the `libbeat/testing` package for unit test helpers. The testing package is used to provide helpers for the `<beatname> test ...` CLI command. By importing `testing` here all Beats will link in the testing package and will have all `-test.X` CLI flags.

Reply:
Looks like it's not easy to create shared test utilities that can be used across packages. I think the next best thing might be to create an [internal package](https://golang.org/doc/go1.4#internalpackages) so at least all code within `libbeat` can access packages under it but any packages outside libbeat cannot. I did this in 34fb1f9. WDYT?

Reply:
Works for me. We have some other test helpers (e.g. check that we don't have any leaky go-routines). Maybe create an issue to 'unify' testing helpers organization.

Reply:
https://github.com/elastic/beats/issues/18239